aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -286,8 +287,10 @@ impl<'e, E: ExecutorView> TDelayedFieldView for StorageAdapter<'e, E> {
         &self,
         delayed_write_set_keys: &HashSet<Self::Identifier>,
         skip: &HashSet<Self::ResourceKey>,
-    ) -> Result<BTreeMap<Self::ResourceKey, (Self::ResourceValue, Arc<MoveTypeLayout>)>, PanicError>
```

Comment:
Why do need the resource size for these reads? To compute the metadata `InPlaceDelayedFieldChange`?

Reply:
We had a write op and were using it for metadata and size, now we pass these directly. Maybe we won't need the size anymore but that's outside the scope here (already huge PR)